### ISSUE TITLE
20240515 Minutes

### ISSUE THREAD

**Issue by sho-i98** [coruubc/diverse_price_prediction#43]
# 20240515 Minutes
Members present: William, Alexi, Sam, Sophia, Gretel, Sho

## Recap and Updates 
**Hong Kong and Taiwan was removed from the dataset**
- As par data quality issue identified in a previous meeting
 
**A Tonnes-weighted price is calculated to aggregate prices within each EZZ, species, and year combination. SubUnits (ports) are removed**

## Progress of this week
**Additional EDA**
- 114 unique EEZ
- Range from 1990 2015
- 0.001 tonnes to 1693787 tonnes
- Growing trend in aquaculture production
  - 5 countries (China, Spain, Indonesia, Japan, Thailand) account for closer to 80% of the production
- Growing trend in consumption likewise
  - More evenly distributed among all countries
- Some correlated features between GDP/Population/SeafoodConumption

**Detrending Methods**
- Various detrending methods explored, including:
 - Differencing
 - Moving average
 - Log-transformation

**Modeling**
- 75% of the whole data set to train and validate, leave out 25% as test set. Wihtin in the 75% of the data, we will use 80% to train, and the 20% to validate in Cross Validation.

**Linear Models**
- The best performing Multiple Linear Regression model was when using Standard Scaling data 
  - Explanatory variables included: EEZID, Species, Production, GDP, Population
  - RMSE Train 36.57, Valid: 37.78, Test: 19.28
  - MAE Train 5.99, Valid: 6.43, Test: 5.23
  - SeafoodConsumption is dropped as per principal of parsimony (The model was not improving much by adding SeafoodConsumption, so we prefer simpler model instead)
  - Year is also dropped because we cannot assume monotonious increase/decrease as time goes by 
  - Regularization not effective because of small number of explanatory variables we have 
  - Species, EEZ, Pop, GDP

- Coefficients Analysis
  - Species being the leading factor impacting the Price

- Non-linear relationship 
  - Interaction term is introduced, but not the non-linearity yet --> To be introduced in a Machine Learning models for the next meeting.

**Linear Mixed-effect models**
- Linear Model framework, but comprises of Fixed effects (Ovearll data) and Random effects (EEZ/Species combination specific data)
  - Advantageous for explaining the variables dependencies
- R is used as instead of Python.for this model only due to the package implementation
- Log/Standard Scaling data explored, log-transofrmation yields the best results. (Unlike Multiple Linear Regression wiht Standard Scaling) 
- Descriptancy in Train RMSE and Validation RMSE, to investigate further with the faculty mentor
- Computation time for training varies from 11 seconds to 40 seconds

**Discussion**
- Prediction accuracy remains as the partner priority, but some interpretability is also useful when demonstrating the results
- Advanced Machine Learning model to excel in performance at the cost of interpretabilities
- Acceptable computation time for the NN models: Longer fitting time is acceptable, but the prediction time is more of the concern for the partner --> prediction time is much shorter compared to the fitting time. To explore when fitting the ML models
- Any avialable computing power at the Institute --> Digital Alliance Canada as computing platform with GPU, and there is also a Desktop PC that our group can utilize.

## Summary
- EDA results aligns with what the partner expected
- Simple statistic models as a tool for deeper data understanding and some interpretabilities 
- Gradiant Boosted Tree, NN to be developed for the next week
- Working in Institute --> Once a week face-to-face work day where computation power is available

## Action items/expectations for next week 
- [ ] Continue working on Linear Mixed-Effect model for cross- effects
- [ ] Log the price and compare the model performance
- [ ] Gradiant Boosted Tree (Excel in hadling tabular data, involve optimization and tuning)
- [ ] Neural Network that mimic the current elm.fast to compare the performance
- [ ] Message William about in-person work arrangement by the end of this week
- [ ] Send agenda for the next week by Monday (May 20) EOD

**Next week meetings: Tuesday (May 21) 9:00 am -10:00 am**
